### PR TITLE
1425 adding basic environments CRUD

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds the very basic environments CRUD to the backend services.
We still need to add the following things to complete the ticket requirements:
1. Include the definition files for environments
2. For test runs, the environment snapshot should be stored

## Changes

- Open API spec changes
- Models, repositories, and services to support Environments
- DB Migrations

## Fixes

- https://github.com/kubeshop/tracetest/issues/1425 (Partially)

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: server/model/tests.go [hunk 1]
```diff
@@ -103,6 +103,9 @@ type (
 		Fail          int
 
 		Metadata RunMetadata
+
+		// environment
+		EnvSnapshot []EnvironmentValue
```
[the remark this answers is not in the document]
Comment (xoscar) [reply server/model/tests.go:108]: Maybe I'm losing the plot haha so please keep me honest if you think that it doesn't makes sense 😄

[the remark this answers is not in the document]
Comment (mathnogueira) [reply server/model/tests.go:109]: > An environment can be completely removed and recreated with the same name

I don't think that would be a problem, to be honest. We would still have the variables available at the moment of the run. As runs are immutable, that is more of a debug tool than anything else.

File: server/http/controller.go [hunk 2]
```diff
@@ -483,6 +485,20 @@ func metadata(in *map[string]string) model.RunMetadata {
 	return model.RunMetadata(*in)
 }
 
+func environment(ctx context.Context, testDB model.Repository, environmentId string) model.Environment {
+	if environmentId != "" {
+		environment, err := testDB.GetEnvironment(ctx, environmentId)
+
+		if err != nil {
+			return model.Environment{}
+		}
```
Comment (mathnogueira) [inline server/http/controller.go:494]: We have a lot of different errors that can happen here (invalid SQL syntax, connection errors, etc) that we need to handle. I'd suggest adding an error return as well, but that would totally kill the reason for this function to exist. Maybe that logic could be done in the `RunTest` function instead. So you can handle it using `handleDBError`

File: server/http/controller.go [hunk 3]
```diff
@@ -514,3 +530,89 @@ func (c *controller) executeTest(ctx context.Context, test model.Test, runInfo o
...
+func (c *controller) DeleteEnvironment(ctx context.Context, environmentId string) (openapi.ImplResponse, error) {
+	environment, err := c.testDB.GetEnvironment(ctx, environmentId)
+	if err != nil {
+		return handleDBError(err), err
+	}
+
+	err = c.testDB.DeleteEnvironment(ctx, environment)
+	if err != nil {
+		return openapi.Response(http.StatusInternalServerError, err.Error()), err
+	}
```
Comment (mathnogueira) [inline server/http/controller.go:572]: Maybe you should use `handleDBError` instead.

**Review by mathnogueira — COMMENTED** [review]
Very good work, @xoscar! Just left a couple of comments regarding error handling.

File: server/model/tests.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (xoscar) [reply server/model/tests.go]: This is done 😄 I added the environment json as snapshot to the test run table

File: server/http/controller.go (hunk 2 above)
Comment (xoscar) [reply server/http/controller.go]: I added the error handling, let me know if it makes sense

File: server/http/controller.go (hunk 3 above)
Comment (xoscar) [reply server/http/controller.go]: Done 😄

**Review by mathnogueira — APPROVED** [review]
LGTM. I really liked the db tests, they make it very easy to review the db code :D

**Pull request merged**